bindings/rust/src/lib.rs

```diff
@@ -40,6 +40,32 @@ mod sys;
 use std::ffi::{CStr, CString};
 use std::ptr::NonNull;
 
+/// The various kinds of errors, which can be returned by any of the interfaces.
+#[derive(Eq, PartialEq, Debug, Clone)]
+pub enum Error {
+    MalformedModule(String),
+    InvalidModule(String),
+    InstantiationFailed(String),
+    Other(String),
+    FunctionNotFound,
```

Comment:
The top 4 errors are 1-to-1 match with the C ones, and the below 5 are Rust specific.